stellargraph/core/element_data.py

```diff
@@ -303,74 +303,117 @@ def _numpyise(d, dtype):
     return {k: np.array(v, dtype=dtype) for k, v in d.items()}
 
 
+class AdjList:
```

Comment:
It's minor, but I think this isn't quite an adjacency list, but actually something like:

```suggestion
class LazyAdjLists:
```